cranelift-filetests/src/function_runner.rs

```diff
@@ -76,13 +179,175 @@ impl FunctionRunner {
         };
 
         let code_page = code_page.make_exec().map_err(|e| e.to_string())?;
-        let callable_fn: fn() -> bool = unsafe { mem::transmute(code_page.as_ptr()) };
 
-        // execute
-        if callable_fn() {
-            Ok(())
-        } else {
-            Err(format!("Failed: {}", context.func.name.to_string()))
+        // Merge two i64 to i128 if necessary.
+        let ret_value_type = return_type(&context.func)?;
+
+        match &self.action {
+            FunctionRunnerAction::Test => {
+                if !ret_value_type.is_bool() {
+                    return Err(String::from("Function return type is not boolean"));
+                }
+                let result = Self::invoke::<bool>(&code_page);
+                if result {
+                    return Ok(());
+                }
+                return Err(format!("Failed: {}", context.func.name.to_string()));
+            }
+            FunctionRunnerAction::Print => Self::check(
+                code_page,
+                ret_value_type,
+                Operator::EQ,
+                ConstantData::default().append(1),
+                true,
+            ),
```

Comment:
Perhaps we should not overload `check`; why not just `invoke` here and `println!` the returned value?